Add unit test specific to Address utils (#1251)

jbogacz opened the pull request:
<!-- 0. 🎉 Thank you for submitting a PR! -->

<!-- 1. **Does this close any open issues?** If so, list them here. If not, remove the `Fixes #` line. -->

# 🚀 Description

<!-- 2. Describe the changes introduced in this pull request -->
<!--    Include any context necessary for understanding the PR's purpose. -->

* Added AddressImpl as contract proxy to invoke and test Address library
* 'Is account address' verified on truffle 'creator' variable
* 'Is contract address' verified on SimpleToken instance deployed in truffle testing context

<!-- 3. Before submitting, please review the following checklist: -->

- [+] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [+] ✅ I've added tests where applicable to test my new functionality.
- [+] 📖 I've made sure that my contracts are well-documented.
- [+] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

nventuro commented on test/Address.test.js:
```diff
@@ -0,0 +1,22 @@
+require('chai')
+  .should();
+
+contract('Address', function ([_, creator]) {
```
nit: I'd rename `creator` to `anyone`, since we don't really need a creator here.

nventuro commented on test/Address.test.js:
```diff
@@ -0,0 +1,22 @@
+  });
+
+  it('should return false for account address', async function () {
+    const isContract = await this.mock.isContract(creator);
```
Could you replace this (and the other test) for `(await this.mock.isContract(creator)).should.equal(false);`? See #1216

nventuro commented on test/Address.test.js:
```diff
@@ -0,0 +1,22 @@
+  });
+
+  it('should return true for contract address', async function () {
+    const contract = await SimpleToken.new({ from: creator });
```
This `from: creator` isn't really needed, why is why I suggested the rename, and removing this bit (to avoid unnecessary confusion about a creator address) :)

jbogacz commented:
@nventuro  be my guest :) I need some time to learn OZ flavors but I'm determined :) Anyway, I've fixed your review comments. I've squashed my commits and --force pushed to my branch - hope it's fine with OZ policy

jbogacz commented:
I'm not sure what happened why 'static test' failed with message 'No output has been received in the last 10m0s'. Does it mean that job maximum 10 minutes was reached and Travis killed it automatically? If so, could you @nventuro kick it again, please?

nventuro commented:
@jbogacz I've seen that happen in a couple suite runs, not sure if there's something wrong with our suite (e.g. a missing `await`) that cuases that - a rerun usually works.

Regarding reviews, it's usually better to simply add commits on top, since we squash before merging anyway, and it makes reviewing easier, but for small PRs like this one it's fine either way :)
